flink-connector-aws/flink-connector-sqs/src/main/java/org.apache.flink.connector.sqs/sink/SqsSinkWriter.java

```diff
@@ -0,0 +1,272 @@
+                        requestResult.accept(Collections.emptyList());
+                    }
+                });
+    }
+
+    @Override
+    protected long getSizeInBytes(SendMessageBatchRequestEntry requestEntry) {
+        try {
+            return requestEntry.messageBody().getBytes("UTF-8").length;
+        } catch (UnsupportedEncodingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void close() {
+        AWSGeneralUtil.closeResources(httpClient, sqsAsyncClient);
+    }
+
+    private void handleFullyFailedRequest(
+            Throwable err, List<SendMessageBatchRequestEntry> requestEntries, Consumer<List<SendMessageBatchRequestEntry>> requestResult) {
+
+        numRecordsOutErrorsCounter.inc(requestEntries.size());
+        boolean isFatal = SQS_EXCEPTION_HANDLER.consumeIfFatal(err, getFatalExceptionCons());
+        if (isFatal) {
+            return;
+        }
+
+        if (failOnError) {
+            getFatalExceptionCons()
+                    .accept(new SqsSinkException.SqsFailFastSinkException(err));
+            return;
+        }
+
+        LOG.warn(
+                "SQS Sink failed to write and will retry {} entries to SQS first request was {}",
+                requestEntries.size(),
+                requestEntries.get(0).toString(),
+                err);
+        requestResult.accept(requestEntries);
+    }
+
+    private void handlePartiallyFailedRequest(
+            SendMessageBatchResponse response,
+            List<SendMessageBatchRequestEntry> requestEntries,
+            Consumer<List<SendMessageBatchRequestEntry>> requestResult) {
+
+        if (response.failed() != null) {
+            LOG.debug("handlePartiallyFailedRequest: SQS Sink failed to write and will retry {} entries to SQS",
+                    response.failed().size());
+            numRecordsOutErrorsCounter.inc(response.failed().size());
+        }
+
+        if (failOnError) {
+            getFatalExceptionCons()
+                    .accept(new SqsSinkException.SqsFailFastSinkException());
+            return;
+        }
+
+        if (response.failed() != null) {
+            final List<SendMessageBatchRequestEntry> failedRequestEntries =
+                    new ArrayList<>(response.failed().size());
+
+            for (final BatchResultErrorEntry failedEntry : response.failed()) {
+                final Optional<SendMessageBatchRequestEntry> retryEntry = getFailedRecord(requestEntries, failedEntry.id());
+                if (retryEntry.isPresent()) {
+                    failedRequestEntries.add(retryEntry.get());
+                }
+            }
+
+            requestResult.accept(failedRequestEntries);
+        }
+    }
+
+    private Optional<SendMessageBatchRequestEntry> getFailedRecord(List<SendMessageBatchRequestEntry> requestEntries,
+                                                                   String selectedId)
+    {
+        return requestEntries.stream()
```

Comment:
Could we avoid java streams as per [coding standards](https://flink.apache.org/how-to-contribute/code-style-and-quality-java/#java-streams)

Reply:
sure. Removed